File: Longest Consecutive Sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0){
            return 0;
        }
        
        Set<Integer> pool = new HashSet<>();
        
        for (int i = 0; i < nums.length; i ++) {
            pool.add(nums[i]);
        }
        
        int ans = 0; 
        for (int i = 0; i < nums.length; i ++){
            if (!pool.contains(nums[i])) {
                continue;
            }
            int streak = 1, x = nums[i];
            pool.remove(x);
            int l = x - 1, r = x + 1;
            while (pool.contains(l)){
                pool.remove(l);
                streak ++;
                l --;
            }
            
            while (pool.contains(r)){
                pool.remove(r);
                streak ++;
                r ++;
                
            }
            ans = Math.max(ans, streak);
        }
        return ans;
    }
}

File: Longest Consecutive Sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if (len(nums) == 0):
            return 0
        
        pool = set(nums);
        maxStreak = 0;
        
        for num in nums:
            if num not in pool:
                continue;
            
            # num in pool
            # a new streak
            streak = 1;
            pool.remove(num)
            l = num - 1;
            r = num + 1;
            while l in pool:
                pool.remove(l)
                streak += 1;
                l -= 1 
                
            while r in pool:
                pool.remove(r);
                streak += 1;
                r += 1;
                
            maxStreak = max(maxStreak, streak);
            
            
        return maxStreak;
            
                
        
        